opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -291,23 +291,25 @@ def get_aggregated_resources(
     :param timeout: Number of seconds to wait for each detector to return
     :return:
     """
-    final_resource = initial_resource or _EMPTY_RESOURCE
-    detectors = [OTELResourceDetector()] + detectors
+    detectors_merged_resource = initial_resource or _EMPTY_RESOURCE
 
     with concurrent.futures.ThreadPoolExecutor(max_workers=4) as executor:
         futures = [executor.submit(detector.detect) for detector in detectors]
         for detector_ind, future in enumerate(futures):
             detector = detectors[detector_ind]
             try:
-                detected_resources = future.result(timeout=timeout)
+                detected_resource = future.result(timeout=timeout)
             # pylint: disable=broad-except
             except Exception as ex:
                 if detector.raise_on_error:
                     raise ex
                 logger.warning(
                     "Exception %s in detector %s, ignoring", ex, detector
                 )
-                detected_resources = _EMPTY_RESOURCE
+                detected_resource = _EMPTY_RESOURCE
             finally:
-                final_resource = final_resource.merge(detected_resources)
-    return final_resource
+                detectors_merged_resource = detectors_merged_resource.merge(
+                    detected_resource
+                )
+
+    return Resource.create().merge(detectors_merged_resource)
```

Comment:
That's true but that was my intention.

First, because the [spec says](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/resource/semantic_conventions#service) you MUST set the `service.name` anyways.

Second, because I think it's okay to make sure very span has

```python
{
    'telemetry.sdk.language': "python",
    'telemetry.sdk.name': "opentelemetry",
    'telemetry.sdk.version': _OPENTELEMETRY_SDK_VERSION,
}
```

I can't think of any reason why someone would _not_ want this on their resource.

Reply:
> First, because the [spec says](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/resource/semantic_conventions#service) you MUST set the `service.name` anyways.

It's actually ok as long as it's not the default, check out [here in the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#:~:text=Note%3A%20This%20means,associating%20a%20resource.)

I agree that the default should include all that stuff. However, someone may want to explicitly turn it off. This PR is fixing what we consider a bug, so it's ok that it changes users' behavior. However, it would nice that if anyone preferred it the way it was, they can get back that behavior.

Is there a problem with making this possible and not the default? I don't see a downside

Reply:
> It's actually ok as long as it's not the default, check out here in the spec

I can't find the part that says it's ok, but I'll take your word for it 😅 

> I agree that the default should include all that stuff. However, someone may want to explicitly turn it off. This PR is fixing what we consider a bug, so it's ok that it changes users' behavior. However, it would nice that if anyone preferred it the way it was, they can get back that behavior.

> Is there a problem with making this possible and not the default? I don't see a downside

Hm I have no complaints with this 🙂 I think it _should_ be the default but that the default can also be overridden, which is what your solution gives, so I like it a lot and will do that 😄

Reply:
Ahh the link to text may not have worked. Here is the text fro that link:

> Note: This means that it is possible to create and associate a resource that does not have all or any of the SDK-provided attributes present. However, that does not happen by default. If a user wants to combine custom attributes with the default resource, they can use Merge with their custom resource or specify their attributes by implementing Custom resource detectors instead of explicitly associating a resource.

Reply:
Nice, with the change you suggested, now the "default" is to have the attributes there, and "it is possible to create... a resource that does not have...the SDK-provided attributes". So marking as resolved!